src/gl.rs

```diff
@@ -54,6 +54,7 @@ fn calculate_length(width: GLsizei, height: GLsizei, format: GLenum, pixel_type:
     };
     let depth = match pixel_type {
         ffi::UNSIGNED_BYTE => 1,
+        ffi::FLOAT=> 4,
```

Comment:
shouldn't we also be checking `format` for `ffi::DEPTH`?

Reply:
why? ```depth = match pixel_type```only gets the bytes per color component. The ```match format``` gets the number of color components.

Reply:
What do you pass when reading the depth? I'd expect `format == ffi::DEPTH` and `pixel_type == ffi::FLOAT`

Reply:
ok, I added the DEPTH format check. According to the spec glReadPixels uses ffi::DEPTH_COMPONENT instead of ffi::DEPTH